### ISSUE TITLE
Is there any way to support kinematic loop

### ISSUE THREAD

**Issue by lvitol** [Genesis-Embodied-AI/Genesis#22]
The robot I would like to simulate has the following design:
![image](https://github.com/user-attachments/assets/3cd7565e-01db-4cfa-addc-1c13af8a4a69)

The current code seems to support only tree-like structures.

**Comment by zswang666** [comment]
Yes you are right. Current rigid solver does not support kinematic loop. We will see if there is any workaround for parallel robots and if not, discuss internally when to integrate this. Will keep you posted.

**Comment by YilingQiao** [comment]
We now support equality constraint recently and kinematic loop. There might be some bugs with closed loop + ball joint and we will fix it soon.
